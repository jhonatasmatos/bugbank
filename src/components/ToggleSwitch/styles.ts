import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
