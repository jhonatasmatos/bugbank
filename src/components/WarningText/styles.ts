import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
