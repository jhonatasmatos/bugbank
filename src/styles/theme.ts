export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
