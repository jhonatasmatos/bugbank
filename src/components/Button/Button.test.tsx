import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import { Button } from '.';

const fn = jest.fn();

describe('<Button />', () => {
  it('should render with success', () => {
    const { container, getByText } = renderTheme(
      <Button label="Enviar" onClick={fn} />,
    );

    const button = getByText('Enviar');

    fireEvent.click(button);

    expect(container).toBeInTheDocument();
    expect(fn).toHaveBeenCalledTimes(1);
    expect(button).toHaveTextContent('Enviar');
  });

  it('should render pink button', () => {
    const { container } = renderTheme(
      <Button label="Enviar" appearance="pink" />,
    );

    const button = container.querySelector('button');

    expect(button).toHaveStyle(`background-color: ${theme.colors.primary}`);
    expect(button).toHaveStyle(`color: ${theme.colors.white}`);
  });

  it('should render white button', () => {
    const { container } = renderTheme(
      <Button label="Enviar" appearance="white" />,
    );

    const button = container.querySelector('button');

    expect(button).toHaveStyle(`background-color: ${theme.colors.white}`);
    expect(button).toHaveStyle(`color: ${theme.colors.secondary}`);
  });

  it('should render purple button', () => {
    const { container } = renderTheme(
      <Button label="Enviar" appearance="purple" />,
    );

    const button = container.querySelector('button');

    expect(button).toHaveStyle(`background-color: ${theme.colors.secondary}`);
    expect(button).toHaveStyle(`color: ${theme.colors.white}`);
  });

  it('should render with success', () => {
    const { container } = renderTheme(<Button label="Enviar" onClick={fn} />);
    expect(container).toMatchSnapshot();
  });
});
