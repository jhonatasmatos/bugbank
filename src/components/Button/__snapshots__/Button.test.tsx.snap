// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with success 1`] = `
.c0 {
  -webkit-filter: drop-shadow(2px 2px 4px rgba(0,0,0,0.4));
  filter: drop-shadow(2px 2px 4px rgba(0,0,0,0.4));
  width: 100%;
  padding: 1.6rem 0;
  border-radius: 0.8rem;
  cursor: pointer;
  outline: none;
  font-size: 1.6rem;
  font-family: 'Nunito',sans-serif;
  color: #fff;
  background: #D62A92;
  border: 1px solid #D62A92;
}

.c0:hover {
  opacity: 0.8;
}

<div>
  <button
    class="c0 button__child"
    type="button"
  >
    Enviar
  </button>
</div>
`;
